docs/feature/vector/index.md

```diff
@@ -0,0 +1,274 @@
+(hnsw)=
+(vector)=
+
+# Vector Store
+
+:::{include} /_include/links.md
+:::
+:::{include} /_include/styles.html
+:::
+
+**Vector search on machine learning embeddings: CrateDB is all you need.**
+
+:::::{grid}
+:padding: 0
+
+::::{grid-item}
+:class: rubric-slim
+:columns: auto 9 9 9
+
+
+:::{rubric} Overview
+:::
+CrateDB can be used as a [vector database] (VDBMS) for storing and retrieving
+vector embeddings based on the FLOAT_VECTOR data type and its accompanying
+KNN_MATCH function, effectively conducting HNSW semantic similarity searches
+on them, also known as vector search.
+
+:::{rubric} About
+:::
+Vector search leverages machine learning (ML) to capture the meaning and
+context of unstructured data, including text and images, transforming it
+into a numeric representation.
+
+Frequently used for semantic search, vector
+search finds similar data using approximate nearest neighbor (ANN) algorithms.
+Compared to traditional keyword search, vector search yields more relevant
+results and executes faster.
+
+:::{rubric} Details
+:::
+CrateDB uses Lucene as a storage layer, so it inherits the implementation
+and concepts of Lucene Vector Search, in the same spirit as Elasticsearch.
+
+To learn more details about what's inside, please refer to the [HNSW] graph
+search algorithm, [how Lucene implemented it][making of Lucene vector search],
+[how Elasticsearch now also builds on it][Vector search in Elasticsearch],
+and why effectively [Lucene Is All You Need].
+
+While Elasticsearch uses a [query DSL based on JSON], in CrateDB, you can work
+with Lucene Vector Search using SQL.
+::::
+
+
+::::{grid-item}
+:class: rubric-slim
+:columns: auto 3 3 3
+
+```{rubric} Reference Manual
+```
+- [](inv:crate-reference#type-float_vector)
+- [](inv:crate-reference#scalar_knn_match)
+
+```{rubric} Related
+```
+- {ref}`sql`
+- {ref}`fulltext`
+- {ref}`machine-learning`
+- {ref}`query`
+
+{tags-primary}`SQL`
+{tags-primary}`Vector Store`
+{tags-primary}`Machine Learning`
+::::
+
+:::::
+
+
+## Synopsis
+
+Store and query word embeddings using similarity search based on Cosine
+distance.
```

Comment:
CrateDB uses `VectorSimilarityFunction.EUCLIDEAN`, not COSINE. I got that wrong. /cc @ckurze 

-- https://github.com/crate/crate/blob/5.6.3/server/src/main/java/io/crate/types/FloatVectorType.java#L55

Reply:
For certain applications, we need to have the distance value, which is currently not conveyed through the `_score` column that has a different meaning.

- https://github.com/crate/crate/issues/14801
- https://github.com/crate-workbench/langchain/issues/19

Another idea to expand the corresponding features could be to also unlock the other algorithms from Lucene's [org.apache.lucene.index.VectorSimilarityFunction](https://lucene.apache.org/core/9_9_1/core/org/apache/lucene/index/VectorSimilarityFunction.html), like COSINE, DOT_PRODUCT, and MAXIMUM_INNER_PRODUCT.

/cc @geragray

Reply:
A few more references about that topic on the crate/crate issue tracker.

- https://github.com/crate/crate/issues/6910
- https://github.com/crate/crate/issues/14801
- https://github.com/crate/crate/issues/15768
- https://github.com/crate/crate/issues/15835

Now that the euclidean distance function is implemented, the page should include an example how to use it in a sensible way.

- https://github.com/crate/crate/pull/15832

Maybe one of you can support here, @surister, @simonprickett, @hlcianfagna, @hammerhead, @BaurzhanSakhariev?

Reply:
> Now that the euclidean distance function is implemented, the [relevant page](https://cratedb-guide--53.org.readthedocs.build/feature/vector/) should include an example how to use it in a sensible way.

I've added a note to the backlog, so it will not be forgotten. Any support on this is much welcome.

- GH-101